proposals/0041-i18n-routing.md

```diff
@@ -0,0 +1,377 @@
+- `prependWith?: string`: a path to prepend to `locale`
+- `normalizeLocale?: boolean`: when `true`, the locale is transformed in lower case and the underscore (`_`) is replaced with dash (`-`) 
+
+### Fallback control
+
+An option that tells Astro what to do in case it doesn't find a page that belongs to one of the `locales` configured.
+
+The option `fallbackControl` is a string that can have two values:
+
+- `none`, the **default** Astro should not do anything, and a 404 is rendered;
+- [`redirect`](#redirect), Astro should use a redirect when re-routing to the destination locale; when `defaultLocale` is set, Astro will create a redirect to `defaultLocale` 
+
+The logic is triggered **only** when the path `/` is visited.
+
+
+### Fallback system
+
+The fallback system is a feature that allows users to re-route users from one locale to another in case a page is missing.
+
+The fallback system is an opt-in feature.
+
+The fallback system is configured using `fallback`. `fallback` is an object where both keys and values must be locales.
+
+The key is the locale that should benefit from the fallback system and the value is the locale where Astro should re-route.
+
+Astro will throw an error if any locale in `fallback` (keys and values) isn't present in the `locales` list.
+
+
+#### Redirect
+
+In the example below, we introduce a `pt_BR` in the `locales` list, and we tell Astro via the `fallback` key that `pt_BR` should fall back to the `pt` locale.  We use the `fallbackControl` option to tell Astro that it must apply a redirect for missing pages that belong to the `pt_BR` locale.
+
+```js
+// astro.config.mjs
+import {defineConfig} from "astro/config"
+export default defineConfig({
+    i18n: {
+        defaultLocaLe: 'en',
+        locales: ['en', 'es', 'pt_BR', 'pt', 'fr'],
+        fallback: {
+            pt_BR: 'pt'
+        },
+        fallbackControl: "redirect"
+    }
+})
+```
+
+Let's suppose a user navigates to `/pt-br/welcome`, and the developer **didn't** create the route `src/pages/pt_BR/welcome.*`, Astro will attempt to **redirect** said user to the `/pt/welcome` URL. If the developer **does** have `src/pages/pt/welcome.*` in their file system, the user will see the page correctly, **otherwise** the user will see the 404 page.
+
+### Browser detection
+
+> [!NOTE]
+> This feature requires adapter support using an Astro feature
+
+An opt-in feature that allows to detect the `Accept-Langauge` header sent by the browser, and redirect users the corresponding locale. If a locale isn't supported - AKA isn't present in the `locales` list - Astro won't execute any redirect.
+
+The user can enable the feature using the `detectBrowserLanguage` configuration:
+
+```js
+// astro.config.mjs
+import {defineConfig} from "astro/config"
+export default defineConfig({
+    i18n: {
+        defaultLocaLe: 'en',
+        locales: ['en', 'es', 'pt_BR', 'pt', 'fr'],
+        detectBrowserLangauge: true
```

Comment:
bikeshed: Should it be `detectLanguage` or something else? Not necessarily coming from a browser.